src/bluesky/callbacks/tiled_writer.py

```diff
@@ -66,137 +96,181 @@ def __init__(self, client: BaseClient):
-            df_dict = {"descriptor_uid": uid}
-            df_dict.update(det_dict.get("data", {}))
-            df_dict.update({f"ts_{c}": v for c, v in det_dict.get("timestamps", {}).items()})
-            df = pd.DataFrame(df_dict, index=[0], columns=df_dict.keys())
-            if det_name in conf_node.keys():
-                conf_node[det_name].append_partition(df, 0)
-            else:
-                conf_node.new(
-                    structure_family=StructureFamily.table,
-                    data_sources=[
-                        DataSource(
-                            structure_family=StructureFamily.table,
-                            structure=TableStructure.from_pandas(df),
-                            mimetype="text/csv",
-                        ),
-                    ],
-                    key=det_name,
-                    metadata=det_dict["data_keys"],
-                )
-                conf_node[det_name].write_partition(df, 0)
+        data_keys = doc.get("data_keys", {})
+        self.data_keys_int.update({k: v for k, v in data_keys.items() if "external" not in v.keys()})
+        self.data_keys_ext.update({k: v for k, v in data_keys.items() if "external" in v.keys()})
+        for item in self.data_keys_ext.values():
+            item["external"] = item.pop("external", "")  # Make sure the value set to external key is not None
+
+        # Create a new Composite node for the stream if it does not exist
+        desc_name = doc["name"]  # Name of the descriptor/stream
+        if desc_name not in self.streams_node.keys():
+            extra = {
+                k: v
+                for k, v in doc.items()
+                if k not in {"name", "data_keys", "configuration", "uid", "time", "object_keys", "run_start"}
+            }
+            metadata = data_keys | {
+                "uid": doc["uid"],
+                "time": doc["time"],
+                "extra": extra,
+            }
+            if conf_meta := doc.get("configuration"):
+                metadata.update({"configuration": conf_meta})
+            desc_node = self.streams_node.create_composite(
+                key=desc_name,
+                metadata=metadata,
+                specs=[Spec("BlueskyEventStream", version="3.0")],
+            )
+        else:
+            # Rare Case: This new descriptor likely updates stream configs mid-experiment
+            # We assume tha the full descriptor has been already received, so we don't need to store everything
+            # but only the uid, timestamp, and also data and timestamps in configuration (without conf specs).
+            desc_node = self.streams_node[desc_name]
+            revisions = desc_node.metadata.get("revisions", []) + [{"uid": doc["uid"], "time": doc["time"]}]
+            if conf_meta := doc.get("configuration"):
+                revisions[-1].update({"configuration": conf_meta})
+            desc_node.update_metadata(metadata={"revisions": revisions})
```

Comment:
I really like this solution for dealing with the multiple-descriptor edge case.

I wonder the name `revisions` will cause confusion between `x.metadata_revisions` and `x.metadata["revisions"]`. Maybe `configuration_updates`?